pkg/apis/auditregistration/types.go

```diff
@@ -104,11 +102,37 @@ type Policy struct {
 	Level Level
 
 	// Stages is a list of stages for which events are created.
+	// Must be non-empty if level != none
+	// +optional
+	Stages []Stage
+
+	// Rules define how classes should be handled.
+	// A request may fall under multiple audit classes.
+	// Rules are evaluated in order (first matching wins).
+	// Rules override the top-level & stage.
```

Comment:
You might want to remove the dash in `top-level`. Might be just me, but I didn't realize you refer to the `Level` and `Stage` and that `top` is `Policy` and wondered, which are those `top-level` rules?   
Using capital first letter for `stage` and `level`, referring to the concrete members would make it even more explicit.